diff --git a/grub-core/commands/blsuki.c b/grub-core/commands/blsuki.c
index 4133d31..e5151cd 100644
--- a/grub-core/commands/blsuki.c
+++ b/grub-core/commands/blsuki.c
@@ -1025 +1025 @@ bls_create_entry (grub_blsuki_entry_t *entry)
-  grub_normal_add_menu_entry (argc, argv, classes, id, users, hotkey, NULL, src, 0, entry);
+  grub_normal_add_menu_entry (argc, argv, classes, id, users, hotkey, NULL, src, 0, 0, entry);
@@ -1091 +1091 @@ uki_create_entry (grub_blsuki_entry_t *entry)
-  grub_normal_add_menu_entry (1, argv, NULL, id, NULL, NULL, NULL, src, 0, entry);
+  grub_normal_add_menu_entry (1, argv, NULL, id, NULL, NULL, NULL, src, 0, 0, entry);
diff --git a/grub-core/commands/legacycfg.c b/grub-core/commands/legacycfg.c
index 7d9f9eb..85129a0 100644
--- a/grub-core/commands/legacycfg.c
+++ b/grub-core/commands/legacycfg.c
@@ -146 +146 @@ legacy_file (const char *filename)
-					entrysrc, 0, NULL);
+					entrysrc, 0, 0, NULL);
@@ -207 +207 @@ legacy_file (const char *filename)
-				  NULL, NULL, entrysrc, 0, NULL);
+				  NULL, NULL, entrysrc, 0, 0, NULL);
diff --git a/grub-core/commands/menuentry.c b/grub-core/commands/menuentry.c
index 5e1318f..e85f790 100644
--- a/grub-core/commands/menuentry.c
+++ b/grub-core/commands/menuentry.c
@@ -81 +81 @@ grub_normal_add_menu_entry (int argc, const char **args,
-			    int submenu, grub_blsuki_entry_t *blsuki)
+			    int submenu, int hidden, grub_blsuki_entry_t *blsuki)
@@ -190,0 +191 @@ grub_normal_add_menu_entry (int argc, const char **args,
+  (*last)->hidden = hidden;
@@ -193 +194,2 @@ grub_normal_add_menu_entry (int argc, const char **args,
-  menu->size++;
+  if (!hidden)
+    menu->size++;
@@ -293,0 +296 @@ grub_cmd_menuentry (grub_extcmd_context_t ctxt, int argc, char **args)
+				       ctxt->extcmd->cmd->name[0] == 'h',
@@ -311 +314,2 @@ grub_cmd_menuentry (grub_extcmd_context_t ctxt, int argc, char **args)
-				  ctxt->extcmd->cmd->name[0] == 's', NULL);
+				  ctxt->extcmd->cmd->name[0] == 's',
+				  ctxt->extcmd->cmd->name[0] == 'h', NULL);
@@ -319 +323 @@ grub_cmd_menuentry (grub_extcmd_context_t ctxt, int argc, char **args)
-static grub_extcmd_t cmd, cmd_sub;
+static grub_extcmd_t cmd, cmd_sub, cmd_hidden;
@@ -334,0 +339,6 @@ grub_menu_init (void)
+  cmd_hidden = grub_register_extcmd ("hiddenentry", grub_cmd_menuentry,
+				     GRUB_COMMAND_FLAG_BLOCKS
+				     | GRUB_COMMAND_ACCEPT_DASH
+				     | GRUB_COMMAND_FLAG_EXTRACTOR,
+				     N_("BLOCK"), N_("Define a hidden menu entry."),
+				     options);
@@ -340,0 +351 @@ grub_menu_fini (void)
+  grub_unregister_extcmd (cmd_hidden);
diff --git a/grub-core/normal/menu.c b/grub-core/normal/menu.c
index b946c83..f5e9038 100644
--- a/grub-core/normal/menu.c
+++ b/grub-core/normal/menu.c
@@ -40,0 +41,2 @@
+#define MENU_INCLUDE_HIDDEN 0x10000
+
@@ -84,2 +86,18 @@ grub_menu_get_entry (grub_menu_t menu, int no)
-  for (e = menu->entry_list; e && no > 0; e = e->next, no--)
-    ;
+  if (no & MENU_INCLUDE_HIDDEN)
+    {
+      no &= ~MENU_INCLUDE_HIDDEN;
+
+      for (e = menu->entry_list; e && no > 0; e = e->next, no--)
+        ;
+    }
+  else
+    {
+      for (e = menu->entry_list; e && no > 0; e = e->next, no--)
+        {
+          /* Skip hidden entries.  */
+          while (e && e->hidden)
+            e = e->next;
+        }
+      while (e && e->hidden)
+        e = e->next;
+    }
@@ -97 +115 @@ get_entry_index_by_hotkey (grub_menu_t menu, int hotkey)
-  for (i = 0, entry = menu->entry_list; i < menu->size;
+  for (i = 0, entry = menu->entry_list; entry;
@@ -100 +118 @@ get_entry_index_by_hotkey (grub_menu_t menu, int hotkey)
-      return i;
+      return i | MENU_INCLUDE_HIDDEN;
@@ -512,0 +531,4 @@ get_entry_number (grub_menu_t menu, const char *name)
+      /* Skip hidden entries.  */
+      while (e && e->hidden)
+        e = e->next;
+
@@ -523,0 +546,4 @@ get_entry_number (grub_menu_t menu, const char *name)
+
+	  /* Skip hidden entries.  */
+	  while (e && e->hidden)
+	    e = e->next;
diff --git a/grub-core/normal/menu_text.c b/grub-core/normal/menu_text.c
index 9c383e6..2001216 100644
--- a/grub-core/normal/menu_text.c
+++ b/grub-core/normal/menu_text.c
@@ -295,0 +296,4 @@ print_entries (grub_menu_t menu, const struct menu_viewer_data *data)
+
+      /* Skip hidden entries.  */
+      while (e && e->hidden)
+        e = e->next;
diff --git a/include/grub/menu.h b/include/grub/menu.h
index 8d06e84..f7a3020 100644
--- a/include/grub/menu.h
+++ b/include/grub/menu.h
@@ -74,0 +75,2 @@ struct grub_menu_entry
+  int hidden;
+
diff --git a/include/grub/normal.h b/include/grub/normal.h
index d0150e3..7bab20e 100644
--- a/include/grub/normal.h
+++ b/include/grub/normal.h
@@ -148 +148 @@ grub_normal_add_menu_entry (int argc, const char **args, char **classes,
-			    int submenu, grub_blsuki_entry_t *blsuki);
+			    int submenu, int hidden, grub_blsuki_entry_t *blsuki);
