diff -Naur source-old/src/main.cpp source-new/src/main.cpp
--- source-old/src/main.cpp	1969-12-31 21:00:01.000000000 -0300
+++ source-new/src/main.cpp	2022-10-23 22:30:00.463905363 -0300
@@ -376,15 +376,10 @@
     // Platform specific settings
     vymPlatform = QSysInfo::prettyProductName();
 
-#if defined(Q_OS_WINDOWS)
-    // Only Windows 10 has tar for zip and unzip.
-    // Older windows versions not supported.
-    zipToolPath = "tar";
-    unzipToolPath = "tar";
-#else
-    zipToolPath = "/usr/bin/zip";       // FIXME-4 no longer installed per default in openSUSE 15.6
-    unzipToolPath = "/usr/bin/unzip";
-#endif
+    // Nixpkgs-specific hack
+    zipToolPath = "@zipPath@";
+    unzipToolPath = "@unzipPath@";
+
     iconPath = vymBaseDir.path() + "/icons/";
     flagsPath = vymBaseDir.path() + "/flags/";
 
