diff -ruN a/newlib/libc/include/glob.h b/newlib/libc/include/glob.h
--- a/newlib/libc/include/glob.h	2025-11-23 03:32:00.998972284 -0800
+++ b/newlib/libc/include/glob.h	2025-11-23 03:32:01.208972767 -0800
@@ -55,6 +55,13 @@
 #define	GLOB_NOMAGIC	0x0200	/* GLOB_NOCHECK without magic chars (csh). */
 #define	GLOB_QUOTE	0x0400	/* Quote special chars with \. */
 #define	GLOB_TILDE	0x0800	/* Expand tilde names from the passwd file. */
 #define	GLOB_LIMIT	0x1000	/* limit number of returned paths */
+
+#ifndef GLOB_PERIOD
+#define	GLOB_PERIOD	0x2000	/* Require explicit matches for leading '.' */
+#endif
+#ifndef GLOB_ONLYDIR
+#define	GLOB_ONLYDIR	0x8000	/* Return only directories. */
+#endif
 
 /* backwards compatibility, this is the old name for this option */
 #define GLOB_MAXPATH	GLOB_LIMIT
@@ -63,6 +70,14 @@
 #define	GLOB_NOSPACE	(-1)	/* Malloc call failed. */
 #define	GLOB_ABEND	(-2)	/* Unignored error. */
 #define	GLOB_NOMATCH	(-3)	/* No match and GLOB_NOCHECK not set. */
+
+#define	GLOB_ABORTED	GLOB_ABEND
+#ifndef GLOB_NOESCAPE
+#define	GLOB_NOESCAPE	0x4000
+#endif
+#ifndef GLOB_TILDE_CHECK
+#define	GLOB_TILDE_CHECK	0x10000
+#endif
 
 __BEGIN_DECLS
 int	glob(const char *__restrict, int, int (*)(const char *, int), 
